classes/services/PKPSubmissionService.php

```diff
@@ -0,0 +1,646 @@
+		}
+
+		$dispatcher = $request->getDispatcher();
+
+		// Check if the user is an author of this submission
+		$authorUserGroupIds = DAORegistry::getDAO('UserGroupDAO')->getUserGroupIdsByRoleId(ROLE_ID_AUTHOR);
+		$stageAssignmentsFactory = DAORegistry::getDAO('StageAssignmentDAO')->getBySubmissionAndStageId($submission->getId(), null, null, $user->getId());
+
+		$authorDashboard = false;
+		while ($stageAssignment = $stageAssignmentsFactory->next()) {
+			if (in_array($stageAssignment->getUserGroupId(), $authorUserGroupIds)) {
+				$authorDashboard = true;
+			}
+		}
+
+		// Send authors, journal managers and site admins to the submission
+		// wizard for incomplete submissions
+		if ($submission->getSubmissionProgress() > 0 &&
+				($authorDashboard || $user->hasRole(array(ROLE_ID_MANAGER, ROLE_ID_SITE_ADMIN), $submissionContext->getId()))) {
+			return $dispatcher->url(
+				$request,
+				ROUTE_PAGE,
+				$submissionContext->getPath(),
+				'submission',
+				'wizard',
+				$submission->getSubmissionProgress(),
+				array('submissionId' => $submission->getId())
+			);
+		}
+
+		// Send authors to author dashboard
+		if ($authorDashboard) {
+			return $dispatcher->url(
+				$request,
+				ROUTE_PAGE,
+				$submissionContext->getPath(),
+				'authorDashboard',
+				'submission',
+				$submission->getId()
+			);
+		}
+
+		// Send reviewers to review wizard
+		$reviewAssignment = DAORegistry::getDAO('ReviewAssignmentDAO')->getLastReviewRoundReviewAssignmentByReviewer($submission->getId(), $user->getId());
+		if ($reviewAssignment) {
+			return $dispatcher->url(
+				$request,
+				ROUTE_PAGE,
+				$submissionContext->getPath(),
+				'reviewer',
+				'submission',
+				$submission->getId()
+			);
+		}
+
+		// Give any other users the editorial workflow URL. If they can't access
+		// it, they'll be blocked there.
+		return $dispatcher->url(
+			$request,
+			ROUTE_PAGE,
+			$submissionContext->getPath(),
+			'workflow',
+			$stageName?$stageName:'access',
+			$submission->getId()
+		);
+	}
+
+	/**
+	 * Delete a submission
+	 *
+	 * @param $submissionId int
+	 */
+	public function deleteSubmission($id) {
+		Application::getSubmissionDAO()
+				->deleteById((int) $id);
+	}
+
+	/**
+	 * Check if a user can delete a submission
+	 *
+	 * @param $submission Submission|int Submission object or submission ID
+	 * @return bool
+	 */
+	public function canCurrentUserDelete($submission) {
+
+		if (!is_a($submission, 'Submission')) {
+			$submissionDao = Application::getSubmissionDAO();
+			$submission = $submissionDao->getById((int) $submission);
+			if (!$submission) {
+				return false;
+			}
+		}
+
+		$request = Application::getRequest();
+		$context = $request->getContext();
+		$contextId = $context ? $context->getId() : 0;
+
+		$currentUser = $request->getUser();
+		if (!$currentUser) {
+			return false;
+		}
+
+		$canDelete = false;
+
+		// Only allow admins and journal managers to delete submissions, except
```

Comment:
This policy was taken from https://github.com/pkp/pkp-lib/blob/ojs-stable-3_0_2/controllers/grid/submissions/SubmissionsListGridRow.inc.php#L48, I think -- rather than leaving it implemented in two places, that code should be changed to use this helper too.